qemu/tests/hugepage_specify_node.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest.staging import utils_memory
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu allocate hugepage from node.
+    Steps:
+    1) System setup hugepage on host.
+    2) Mount this hugepage to /mnt/kvm_hugepage.
+    3) Boot guest only allocate hugepage from node.
+    4) Check the hugepage used only from node.
+    :params test: QEMU test object.
+    :params params: Dictionary with the test parameters.
+    :params env: Dictionary with test environment.
+    """
+    error_context.context("Get host numa topological structure.", logging.info)
+    host_numa_node = utils_misc.NumaInfo()
+    node_list = host_numa_node.online_nodes
+
+    pre_hugepages_Free = {}
+    for node_id in node_list:
+        utils_memory.drop_caches()
+        error_context.base_context("Check preprocess HugePages Free on host "
+                                   "numa node %s." % node_id, logging.info)
+        node_MemFree = int(host_numa_node.read_from_node_meminfo(node_id,
+                                                                 "MemFree"))
+        if node_MemFree < int(params["mem"]) * 1024:
+            test.cancel("No enough free memory in node %d." % node_id)
+
+        value = int(host_numa_node.read_from_node_meminfo(node_id,
+                                                          "HugePages_Free"))
+        pre_hugepages_Free[node_id] = value
+
+    for node_id in node_list:
+        error_context.base_context("Specify qemu process only allocate "
+                                   "HugePages from node%s." % node_id, logging.info)
+        params["qemu_command_prefix"] = ("numactl --membind=%s" % node_id)
+        hugepage_size = utils_memory.get_huge_page_size()
+        hugepage_num1 = (int(params['mem']) * 1024) // hugepage_size
```

Comment:
Please move above two lines out of 'for' loop too.

Reply:
Corrected